From 87fa0250081cffb6e413fec7957f5ac4add6cdd1 Mon Sep 17 00:00:00 2001
From: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
Date: Tue, 8 Jan 2013 16:34:26 +0100
Subject: [PATCH] set pkg-config *_location directly to install dir

qmake tries to replace all occurrences of the install prefix with a
literal ${prefix}. of course this would mess up build paths which happen
to live under the prefix (this is untypical, but still), which would
break the sed magic in the subsequent install step. as qmake itself has
no use for the .pc files, it's ok to target them directly for the
install dir. we do the same with the include and lib dirs already.

qt5 is already doing the right thing.

Task-number: QTBUG-28807
Change-Id: I82c17284d1dbac4a96845db2ddec9680e251e7a5
Reviewed-by: Davide Pesavento <davidepesa@gmail.com>
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@digia.com>

diff --git a/src/qbase.pri b/src/qbase.pri
index 991c888..feec9d6 100644
--- a/src/qbase.pri
+++ b/src/qbase.pri
@@ -166,23 +166,25 @@ DEFINES *= QT_USE_QSTRINGBUILDER
 
 TARGET = $$qtLibraryTarget($$TARGET$$QT_LIBINFIX) #do this towards the end
 
-qtPrepareTool(QMAKE_LUPDATE, lupdate)
-qtPrepareTool(QMAKE_LRELEASE, lrelease)
-
+QMAKE_MOC_PC = $$[QT_INSTALL_BINS]/moc
 moc_dir.name = moc_location
-moc_dir.variable = QMAKE_MOC
+moc_dir.variable = QMAKE_MOC_PC
 
+QMAKE_UIC_PC = $$[QT_INSTALL_BINS]/uic
 uic_dir.name = uic_location
-uic_dir.variable = QMAKE_UIC
+uic_dir.variable = QMAKE_UIC_PC
 
+QMAKE_RCC_PC = $$[QT_INSTALL_BINS]/rcc
 rcc_dir.name = rcc_location
-rcc_dir.variable = QMAKE_RCC
+rcc_dir.variable = QMAKE_RCC_PC
 
+QMAKE_LUPDATE_PC = $$[QT_INSTALL_BINS]/lupdate
 lupdate_dir.name = lupdate_location
-lupdate_dir.variable = QMAKE_LUPDATE
+lupdate_dir.variable = QMAKE_LUPDATE_PC
 
+QMAKE_LRELEASE_PC = $$[QT_INSTALL_BINS]/lrelease
 lrelease_dir.name = lrelease_location
-lrelease_dir.variable = QMAKE_LRELEASE
+lrelease_dir.variable = QMAKE_LRELEASE_PC
 
 QMAKE_PKGCONFIG_VARIABLES += moc_dir uic_dir rcc_dir lupdate_dir lrelease_dir
 
-- 
1.8.1.1

